src/allotropy/allotrope/schema_parser/path_util.py

```diff
@@ -66,11 +66,13 @@ def get_schema_path_from_reference(reference: str) -> Path:
 
 
 def get_model_file_from_schema_path(schema_path: Path) -> Path:
-    rel_schema_path = get_rel_schema_path(schema_path)
+    rel_schema_path = PureWindowsPath(get_rel_schema_path(schema_path))
     schema_file = rel_schema_path.name
     model_file = schema_file.replace(".schema.json", ".py").replace("-", "_").lower()
-    schema_dir = str(rel_schema_path.parent)
-    model_path = re.sub("/([0-9]+)", r"/_\1", schema_dir.replace("-", "_").lower())
+    model_path = Path(*[
+        re.sub("^([0-9]+)$", r"_\1", part.lower().replace("-", "_"))
+        for part in rel_schema_path.parent.parts
+    ])
     return Path(model_path, model_file)
```

Comment:
is it ok to wrap `Path` in a `Path`?

Reply:
Yup! Path loves being wrapped in Path. It's it's favorite thing